### DISCUSSION TITLE
Picking from objects and sub-objects in the same query

### DISCUSSION THREAD

**Discussion by woolsweater** [mikefarah/yq#1607] [Q&A]
I have a JSON array of objects where I'd like to pick out a few of the fields, including some in nested objects, while preserving the structure. For example, as input 
(in my real data the key names aren't repetitively named like this; it's just for the example):

```json
[
  {
    "pick_1" : 123,
    "ignore_1" : "foo",
    "ignore_2" : false,
    "pick_object" : {
      "pick_o_1" : "bar",
      "pick_o_2" : true,
      "ignore_o_1" : "baz",
      "ignore_o_2" : "quux",
    },
    "pick_2" : "plugh",
    "ignore_3" : "waldo"
  },
  {
    "pick_1" : 123,
    "ignore_1" : "foo",
    "ignore_2" : false,
    "pick_object" : {
      "pick_o_1" : "bar",
      "pick_o_2" : true,
      "ignore_o_1" : "baz",
      "ignore_o_2" : "quux",
    },
    "pick_2" : "plugh",
    "ignore_3" : "waldo"
  }
]
```

And the output I'd like is just all of the "pick*" fields, with the array and nested objects intact:

```json
[
  {
    "pick_1": 123,
    "pick_object": {
      "pick_o_1": "bar",
      "pick_o_2": true
    },
    "pick_2": "plugh"
  },
  {
    "pick_1": 123,
    "pick_object": {
      "pick_o_1": "bar",
      "pick_o_2": true
    },
    "pick_2": "plugh"
  }
]
```

The simplest query I've been able to come up with does a `pick` and then a `del` to drop the other sub-object fields:

```
[ .[] | pick(["pick_1", "pick_object", "pick_2"]) | del(.pick_object.ignore_o_1, .pick_object.ignore_o_2) ]
```

Is there a way to simplify this to use just a `pick`? Or some other alternative that I've completely missed?

**Comment by mikefarah** [comment]
Have a look at using set_path: https://mikefarah.gitbook.io/yq/operators/path#set-path-to-prune-deep-paths

**Reply by woolsweater** [reply]
Excellent, thanks!
